src/libraries/System.Formats.Tar/src/Resources/Strings.resx

```diff
@@ -0,0 +1,252 @@
+  </data>
+  <data name="IO_FileNotFound" xml:space="preserve">
+    <value>Unable to find the specified file.</value>
+  </data>
+  <data name="IO_FileNotFound_FileName" xml:space="preserve">
+    <value>Could not find file '{0}'.</value>
+  </data>
+  <data name="IO_FifoFiles_NotSupported" xml:space="preserve">
+    <value>Creating fifo files is not supported on the current platform.</value>
+  </data>
+  <data name="IO_NotSupported_UnreadableStream" xml:space="preserve">
+    <value>The stream does not support reading.</value>
+  </data>
+  <data name="IO_NotSupported_UnseekableStream" xml:space="preserve">
+    <value>The stream does not support seeking.</value>
+  </data>
+  <data name="IO_NotSupported_UnwritableStream" xml:space="preserve">
+    <value>The stream does not support writing.</value>
+  </data>
+  <data name="IO_PathNotFound_NoPathName" xml:space="preserve">
+    <value>Could not find a part of the path.</value>
+  </data>
+  <data name="IO_PathNotFound_Path" xml:space="preserve">
+    <value>Could not find a part of the path '{0}'.</value>
+  </data>
+  <data name="IO_PathTooLong" xml:space="preserve">
+    <value>The specified file name or path is too long, or a component of the specified path is too long.</value>
+  </data>
+  <data name="IO_PathTooLong_Path" xml:space="preserve">
+    <value>The path '{0}' is too long, or a component of the specified path is too long.</value>
+  </data>
+  <data name="IO_SetLengthRequiresSeekingAndWriting" xml:space="preserve">
+    <value>SetLength requires a stream that supports seeking and writing.</value>
+  </data>
+  <data name="IO_SharingViolation_File" xml:space="preserve">
+    <value>The process cannot access the file '{0}' because it is being used by another process.</value>
+  </data>
+  <data name="IO_SharingViolation_NoFileName" xml:space="preserve">
+    <value>The process cannot access the file because it is being used by another process.</value>
+  </data>
+  <data name="IO_StreamDisposed" xml:space="preserve">
+    <value>Cannot access a disposed stream.</value>
+  </data>
+  <data name="IO_StreamNotEmpty" xml:space="preserve">
+    <value>The stream is not empty.</value>
+  </data>
+  <data name="PlatformNotSupported_SystemFormatsTar" xml:space="preserve">
+    <value>System.Formats.Tar is not supported on this platform.</value>
+  </data>
+  <data name="SetLengthRequiresSeekingAndWriting" xml:space="preserve">
+    <value>SetLength requires a stream that supports seeking and writing.</value>
+  </data>
+  <data name="TarDuplicateExtendedAttribute" xml:space="preserve">
+    <value>The entry '{0}' has a duplicate extended attribute.</value>
+  </data>
+  <data name="TarUnexpectedMetadataEntry" xml:space="preserve">
+    <value>A metadata entry of type '{0}' was unexpectedly found after a metadata entry of type '{1}'.</value>
+  </data>
+  <data name="TarEntriesInDifferentFormats" xml:space="preserve">
+    <value>The archive contains entries in different formats.</value>
```

Comment:
Does this mean the archive is non compliant with spec, or simply that we don't support this situation? might be helpful to clarify in the message.

Reply:
I improved the message. This exception message would show up when the tar reader unexpectedly finds an entry in a different format than the format of all the previous entries.